kodilla-paterns/src/main/java/com/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,39 @@
+package com.patterns.prototype.library;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype<Library> {
+
+    private String name;
+    private Set<Book> books = new HashSet<>();
+
+    public Library(final String name) {
+        this.name = name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library libraryClone = super.clone();
+        libraryClone.books = new HashSet<>();
+        for(Book theBook : books) {
+            libraryClone.getBooks().add(theBook);
```

Comment:
to nie jest glebokie kopiowanie tak naprawde. Wspoldzielisz ten sam obiekt ksiazki miedzy dwoma bibliotekami